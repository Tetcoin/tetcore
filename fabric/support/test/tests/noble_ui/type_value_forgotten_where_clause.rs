#[fabric_support::noble]
mod noble {
	use fabric_support::noble_prelude::{Hooks, PhantomData};
	use fabric_system::noble_prelude::BlockNumberFor;

	#[noble::config]
	pub trait Config: fabric_system::Config
	where <Self as fabric_system::Config>::AccountId: From<u32>
	{}

	#[noble::noble]
	pub struct Noble<T>(PhantomData<T>);

	#[noble::hooks]
	impl<T: Config> Hooks<BlockNumberFor<T>> for Noble<T>
	where <T as fabric_system::Config>::AccountId: From<u32>
	{}

	#[noble::call]
	impl<T: Config> Noble<T>
	where <T as fabric_system::Config>::AccountId: From<u32>
	{}

	#[noble::type_value] fn Foo<T: Config>() -> u32 { 3u32 }
}

fn main() {
}
