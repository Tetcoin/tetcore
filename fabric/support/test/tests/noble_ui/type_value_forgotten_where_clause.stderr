error[E0277]: the trait bound `<T as fabric_system::Config>::AccountId: From<u32>` is not satisfied
  --> $DIR/type_value_forgotten_where_clause.rs:24:34
   |
7  |     pub trait Config: fabric_system::Config
   |               ------ required by a bound in this
8  |     where <Self as fabric_system::Config>::AccountId: From<u32>
   |                                                      --------- required by this bound in `noble::Config`
...
24 |     #[noble::type_value] fn Foo<T: Config>() -> u32 { 3u32 }
   |                                     ^^^^^^ the trait `From<u32>` is not implemented for `<T as fabric_system::Config>::AccountId`
   |
help: consider further restricting the associated type
   |
24 |     #[noble::type_value] fn Foo<T: Config>() -> u32 where <T as fabric_system::Config>::AccountId: From<u32> { 3u32 }
   |                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0277]: the trait bound `<T as fabric_system::Config>::AccountId: From<u32>` is not satisfied
  --> $DIR/type_value_forgotten_where_clause.rs:24:12
   |
7  |     pub trait Config: fabric_system::Config
   |               ------ required by a bound in this
8  |     where <Self as fabric_system::Config>::AccountId: From<u32>
   |                                                      --------- required by this bound in `noble::Config`
...
24 |     #[noble::type_value] fn Foo<T: Config>() -> u32 { 3u32 }
   |               ^^^^^^^^^^ the trait `From<u32>` is not implemented for `<T as fabric_system::Config>::AccountId`
   |
help: consider further restricting the associated type
   |
24 |     #[noble::type_value where <T as fabric_system::Config>::AccountId: From<u32>] fn Foo<T: Config>() -> u32 { 3u32 }
   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0277]: the trait bound `<T as fabric_system::Config>::AccountId: From<u32>` is not satisfied
  --> $DIR/type_value_forgotten_where_clause.rs:24:12
   |
7  |     pub trait Config: fabric_system::Config
   |               ------ required by a bound in this
8  |     where <Self as fabric_system::Config>::AccountId: From<u32>
   |                                                      --------- required by this bound in `noble::Config`
...
24 |     #[noble::type_value] fn Foo<T: Config>() -> u32 { 3u32 }
   |               ^^^^^^^^^^ the trait `From<u32>` is not implemented for `<T as fabric_system::Config>::AccountId`
   |
help: consider further restricting the associated type
   |
24 |     #[noble::type_value] fn Foo<T: Config>() -> u32 where <T as fabric_system::Config>::AccountId: From<u32> { 3u32 }
   |                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
